* Thu Mar 12 2015 Patrick Monnerat <pm@datasphere.ch>
- MinGW defines struct timespec in unistd.h: detect it.
- Prefer it over the definition in pthread.h to avoid possible conflict.

diff -Naurp a/gdb/gnulib/import/m4/time_h.m4 b/gdb/gnulib/import/m4/time_h.m4
--- a/gdb/gnulib/import/m4/time_h.m4	2015-03-09 15:22:52.000000000 +0100
+++ b/gdb/gnulib/import/m4/time_h.m4	2015-03-12 16:31:50.246669200 +0100
@@ -26,7 +26,7 @@ AC_DEFUN([gl_HEADER_TIME_H_BODY],
 ])
 
 dnl Check whether 'struct timespec' is declared
-dnl in time.h, sys/time.h, or pthread.h.
+dnl in time.h, sys/time.h, unistd.h or pthread.h.
 
 AC_DEFUN([gl_CHECK_TYPE_STRUCT_TIMESPEC],
 [
@@ -43,6 +43,7 @@ AC_DEFUN([gl_CHECK_TYPE_STRUCT_TIMESPEC]
 
   TIME_H_DEFINES_STRUCT_TIMESPEC=0
   SYS_TIME_H_DEFINES_STRUCT_TIMESPEC=0
+  UNISTD_H_DEFINES_STRUCT_TIMESPEC=0
   PTHREAD_H_DEFINES_STRUCT_TIMESPEC=0
   if test $gl_cv_sys_struct_timespec_in_time_h = yes; then
     TIME_H_DEFINES_STRUCT_TIMESPEC=1
@@ -59,22 +60,36 @@ AC_DEFUN([gl_CHECK_TYPE_STRUCT_TIMESPEC]
     if test $gl_cv_sys_struct_timespec_in_sys_time_h = yes; then
       SYS_TIME_H_DEFINES_STRUCT_TIMESPEC=1
     else
-      AC_CACHE_CHECK([for struct timespec in <pthread.h>],
-        [gl_cv_sys_struct_timespec_in_pthread_h],
+      AC_CACHE_CHECK([for struct timespec in <unistd.h>],
+        [gl_cv_sys_struct_timespec_in_unistd_h],
         [AC_COMPILE_IFELSE(
            [AC_LANG_PROGRAM(
-              [[#include <pthread.h>
+              [[#include <unistd.h>
               ]],
               [[static struct timespec x; x.tv_sec = x.tv_nsec;]])],
-           [gl_cv_sys_struct_timespec_in_pthread_h=yes],
-           [gl_cv_sys_struct_timespec_in_pthread_h=no])])
-      if test $gl_cv_sys_struct_timespec_in_pthread_h = yes; then
-        PTHREAD_H_DEFINES_STRUCT_TIMESPEC=1
+           [gl_cv_sys_struct_timespec_in_unistd_h=yes],
+           [gl_cv_sys_struct_timespec_in_unistd_h=no])])
+      if test $gl_cv_sys_struct_timespec_in_unistd_h = yes; then
+        UNISTD_H_DEFINES_STRUCT_TIMESPEC=1
+      else
+        AC_CACHE_CHECK([for struct timespec in <pthread.h>],
+          [gl_cv_sys_struct_timespec_in_pthread_h],
+          [AC_COMPILE_IFELSE(
+             [AC_LANG_PROGRAM(
+                [[#include <pthread.h>
+                ]],
+                [[static struct timespec x; x.tv_sec = x.tv_nsec;]])],
+             [gl_cv_sys_struct_timespec_in_pthread_h=yes],
+             [gl_cv_sys_struct_timespec_in_pthread_h=no])])
+        if test $gl_cv_sys_struct_timespec_in_pthread_h = yes; then
+          PTHREAD_H_DEFINES_STRUCT_TIMESPEC=1
+        fi
       fi
     fi
   fi
   AC_SUBST([TIME_H_DEFINES_STRUCT_TIMESPEC])
   AC_SUBST([SYS_TIME_H_DEFINES_STRUCT_TIMESPEC])
+  AC_SUBST([UNISTD_H_DEFINES_STRUCT_TIMESPEC])
   AC_SUBST([PTHREAD_H_DEFINES_STRUCT_TIMESPEC])
 ])
 
diff -Naurp a/gdb/gnulib/import/Makefile.am b/gdb/gnulib/import/Makefile.am
--- a/gdb/gnulib/import/Makefile.am	2015-03-09 15:22:51.000000000 +0100
+++ b/gdb/gnulib/import/Makefile.am	2015-03-12 16:33:36.923323100 +0100
@@ -1525,6 +1525,7 @@ time.h: time.in.h $(top_builddir)/config
 	      -e 's|@''REPLACE_NANOSLEEP''@|$(REPLACE_NANOSLEEP)|g' \
 	      -e 's|@''REPLACE_TIMEGM''@|$(REPLACE_TIMEGM)|g' \
 	      -e 's|@''PTHREAD_H_DEFINES_STRUCT_TIMESPEC''@|$(PTHREAD_H_DEFINES_STRUCT_TIMESPEC)|g' \
+	      -e 's|@''UNISTD_H_DEFINES_STRUCT_TIMESPEC''@|$(UNISTD_H_DEFINES_STRUCT_TIMESPEC)|g' \
 	      -e 's|@''SYS_TIME_H_DEFINES_STRUCT_TIMESPEC''@|$(SYS_TIME_H_DEFINES_STRUCT_TIMESPEC)|g' \
 	      -e 's|@''TIME_H_DEFINES_STRUCT_TIMESPEC''@|$(TIME_H_DEFINES_STRUCT_TIMESPEC)|g' \
 	      -e '/definitions of _GL_FUNCDECL_RPL/r $(CXXDEFS_H)' \
diff -Naurp a/gdb/gnulib/import/time.in.h b/gdb/gnulib/import/time.in.h
--- a/gdb/gnulib/import/time.in.h	2015-03-09 15:22:52.000000000 +0100
+++ b/gdb/gnulib/import/time.in.h	2015-03-12 16:27:33.625798100 +0100
@@ -53,6 +53,8 @@
 # if ! @TIME_H_DEFINES_STRUCT_TIMESPEC@
 #  if @SYS_TIME_H_DEFINES_STRUCT_TIMESPEC@
 #   include <sys/time.h>
+#  elif @UNISTD_H_DEFINES_STRUCT_TIMESPEC@
+#   include <unistd.h>
 #  elif @PTHREAD_H_DEFINES_STRUCT_TIMESPEC@
 #   include <pthread.h>
 /* The pthreads-win32 <pthread.h> also defines a couple of broken macros.  */
